tcpclient/connectionsManagement_test.go

```diff
@@ -13,41 +12,114 @@ import (
 // We really need to refactor this test. We should verify connections do become established,
 // rather than just waiting for a second and finish
 // We should also test "failing" connections, and ensure their status is reported properly
-func TestTcpConnect(t *testing.T) {
-	var numberConnections = 2
+func TestTCPConnectEstablished(t *testing.T) {
 	var host = "127.0.0.1"
 	var port = 55555
 
 	dispatcher := &tcpserver.Dispatcher{make(map[string]*tcpserver.Handler)}
 
-	run := func() {
+	runTCPServer := func() {
+		t.Log("Starting TCP server...")
 		if err := dispatcher.ListenHandlers(port); err != nil {
-			t.Error("Could not start the TCP server", err)
+			t.Fatal("Could not start the TCP server", err)
 			return
 		}
-		t.Log("TCP server started")
 	}
-	go run()
+	go runTCPServer()
+	time.Sleep(1 * time.Second)
 
+	defer func() {
+		err := recover()
+		if err != "sync: negative WaitGroup counter" {
+			t.Fatalf("Unexpected panic: %#v", err)
+		}
+	}()
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	var statusChannel = make(chan Connection, 2)
+	var closeRequest = make(chan bool)
+
+	// We use a different subroutine to be able to reach the closeRequest command
+	t.Log("Initiating TCP Connect")
+	go TCPConnect(1, host, port, &wg, ioutil.Discard, statusChannel, closeRequest)
```

Comment:
Sometimes this do fail, as I think it sends messages to the gochannel before running the wg.done (so this main goroutine, executing the test flow, marks wg.done before than the actual tcpconnect). See:
```
panic: sync: negative WaitGroup counter

goroutine 18 [running]:
sync.(*WaitGroup).Add(0xc4200a0000, 0xffffffffffffffff)
        /usr/local/go/src/sync/waitgroup.go:75 +0x20e
sync.(*WaitGroup).Done(0xc4200a0000)
        /usr/local/go/src/sync/waitgroup.go:100 +0x42
github.com/dachad/tcpgoon/tcpclient.TCPConnect(0x1, 0x646556, 0x9, 0xd903, 0xc4200a0000, 0x73ef20, 0x776e68, 0xc4200a4000, 0xc4200a6000, 0x0, ...)
        github.com/dachad/tcpgoon/tcpclient/_test/_obj_test/connectionsManagement.go:59 +0x111d
created by github.com/dachad/tcpgoon/tcpclient.TestTCPConnectEstablished
        /home/caba/goWork/src/github.com/dachad/tcpgoon/tcpclient/connectionsManagement_test.go:45 +0x38c
FAIL    github.com/dachad/tcpgoon/tcpclient     2.020s
```

Reply:
thks @dcaba , I've not seen this anytime, but I will have a look 🤔

Reply:
I've been able to reproduce this after running it in a loop...
My point, as yours, is that the test routine closes the wg before the tcpclient... so I'm giving some more time to the tcpclient to accomplish this and follow the defined path

Reply:
it's fixed in commit 51b7ab3cabfa4f9dbdc8568227f637340848662f